### ISSUE TITLE
Build error: hdf-eos2 on macOS after update from spack develop 2024/05/09

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack-stack#1113]
**Describe the bug**
See https://github.com/spack/spack/issues/44168. Because of this, we disabled the additional build variants for `met` in the macOS CI scripts in commit https://github.com/JCSDA/spack-stack/pull/1109/commits/28c1135f6cd9e6be4d5c6cc2fc2522788dea5d93, part of https://github.com/JCSDA/spack-stack/pull/1109.

**To Reproduce**
Re-enable the `met` variants and watch in awe.

**Expected behavior**
No build errors.

**System:**
macOS Ventura aarch64 with `apple-clang@14.x.y`.

**Additional context**
We may just want to wait until this is fixed upstream in spack develop by the developer whose updates to hdf-eos2 apparently broke the build. Again, see https://github.com/spack/spack/issues/44168 for more information.

**Comment by climbfuji** [comment]
We believe this is fixed

**Comment by mathomp4** [comment]
> We believe this is fixed

Actually, maybe not? I just tried installing it picking up the latest package from @danrosen25:

https://github.com/spack/spack-packages/pull/2751

First, I didn't have much luck with 2.20v1.00, so I moved the pin to 3.0 but now it seems to fail because spack-stack *explicitly* builds `hdf` as `~fortran` and has for years. I'll keep exploring...

**Comment by mathomp4** [comment]
Wait. `~fortran` might be a red herring. It's like the build is weirdly broken. Testing now.

**Comment by mathomp4** [comment]
Yeah, it was. But this is fun. I had to work with Claude a bit because apparently there were...issues. Maybe Linux was happy, but Darwin wasn't. But I can build hdf-eos2 v3.0 on macOS it seems.

I'll need to make a PR to spack-packages mainline first and have you all look.

**Comment by danrosen25** [comment]
@mathomp4 @climbfuji 
Do we need this line? `depends_on("hdf+fortran", when="+fortran")` in the hdf-eos2 recipe?

**Comment by mathomp4** [comment]
Oh. huh. I mean, i didn't seem to need it, but then...it does seem smart, right? I've added it to the spack-packages PR here:

https://github.com/spack/spack-packages/pull/4548

and I'll put it locally as well. I guess since spack-stack is `hdf ~fortran` then I didn't hit this. (Though at one point I thought my issue was fortran related)
